rotkehlchen/chain/optimism/modules/velodrome/velodrome_cache.py

```diff
@@ -0,0 +1,218 @@
+from rotkehlchen.types import (
+    AddressbookEntry,
+    CacheType,
+    ChainID,
+    ChecksumEvmAddress,
+    SupportedBlockchain,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.optimism.manager import OptimismInquirer
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+VELODROME_SUGAR_V2_CONTRACT = string_to_evm_address('0x7F45F1eA57E9231f846B2b4f5F8138F94295A726')  # Velodrome Finance LP Sugar v2  # noqa: E501
+
+
+class VelodromePoolData(NamedTuple):
+    pool_address: ChecksumEvmAddress
+    pool_name: str
+    gauge_address: Optional[ChecksumEvmAddress]
+    coins: list[ChecksumEvmAddress]
+
+
+def save_velodrome_pools_to_cache(
+        write_cursor: DBCursor,
+        database: 'DBHandler',
+        new_pools: list[VelodromePoolData],
+) -> None:
+    """
+    Stores data about velodrome pools and gauges in the gloabal db cache tables.
+
+    unique_cache table:
+    VELODROME_GAUGE_ADDRESS + {pool address} -> {gauge address}
+
+    general_cache table:
+    VELODROME_POOL_ADDRESS -> {pool address}
+    """
+    db_addressbook = DBAddressbook(db_handler=database)
+    for pool in new_pools:
+        addresbook_entries = [AddressbookEntry(
+            address=pool.pool_address,
+            name=f'Velodrome pool {pool.pool_name}',
+            blockchain=SupportedBlockchain.OPTIMISM,
+        )]
+        if pool.gauge_address is not None:
+            addresbook_entries.append(AddressbookEntry(
+                address=pool.gauge_address,
+                name=f'Velodrome gauge for Velodrome pool {pool.pool_name}',
+                blockchain=SupportedBlockchain.OPTIMISM,
+            ))
+        try:
+            db_addressbook.add_addressbook_entries(
+                write_cursor=write_cursor,
+                entries=addresbook_entries,
+            )
+        except InputError as e:
+            log.debug(
+                f'Velodrome address book names for pool {pool.pool_address} were not added. '
+                f'Probably names were added by the user earlier. {e!s}')
+
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.VELODROME_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        if pool.gauge_address is not None:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.VELODROME_GAUGE_ADDRESS, pool.pool_address),
+                value=pool.gauge_address,
+            )
+
+
+def read_velodrome_pools_and_gauges_from_cache() -> tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]:  # noqa: E501
+    """
+    Reads globaldb cache and returns:
+    - A set of all known addresses of velodrome pools.
+    - A set of all known addresses of velodrome gauges.
+
+    Doesn't raise anything unless cache entries were inserted incorrectly.
```

Comment:
then it can raise something no? write what it is as I don't have clear what do you mean